3scale/porta
# [Service Discovery] Update/resync

**What this PR does / why we need it**:

Introduces the 'refresh' option for APIs created by service discovery. The action perform an update on the `Service` record (and corresponding `ApiDocs::Service`)  by fetching again the latest definitions of the corresponding service in the cluster, eventually changing with a new backend URL and redeploying the proxy configs, and updating the ActiveDocs.

<img width="1679" alt="screen shot 2018-10-11 at 9 57 21 am" src="https://user-images.githubusercontent.com/1842261/46789150-18a50580-cd3c-11e8-91f2-560b0d5534a9.png">


**Which issue(s) this PR fixes** 

Realted to https://github.com/3scale/porta/issues/34

**Verification steps** 

1. Create an API with Service Discovery
2. Navigate to the Definition page of the new API
3. Check the new 'Source: OpenShift' info and 'refresh' option

**Special notes for your reviewer**:

- [x] Make it only save and deploy a new proxy config if the endpoint really changed
- [x] Update an existing ActiveDoc of the API (now it's always creating a new one every time the user clicks 'refresh')
